### PR TITLE
Display individual votes for a specific MP

### LINKED ISSUE #603: Page for single vote by an individual MP
A page just showing how a specific MP voted on an individual vote.  

Taking eg. the information shown in italics at:

http://www.publicwhip.org.uk/division.php?date=2014-06-23&number=12&mpn=Julian_Huppert&mpc=Cambridge&house=commons

and making that the primary content of a page so people can link to an individual vote on TheyWorkForYou. 

Such a page could be simpler than that on PublicWhip as it wouldn't have the full vote description; just the summary from the point of view of the MP in question. 

With links "up" to the MP's main page; voting record; and the page on any policy positions vote relates to. 

A version of this page without an individual MP set could show how the majority of MPs voted. 

See also #37 , #85 and #518 

single-vote-page

### PR DESCRIPTION
**Pull request by struan**
Builds on the work in #1242 to show how a specific MP voted in a vote. Also links that to the MP voting pages. It also highlights if their vote was in agreement or disagreement with the majority of MPs.

screen shot 2017-04-19 at 12 17 03

screen shot 2017-04-19 at 12 18 47

Fixes #603

### CODE REVIEW

File: www/docs/divisions/division.php [hunk 1]
```diff
@@ -18,11 +22,17 @@
     $template = 'divisions/vote';
     $DATA->set_page_metadata($this_page, 'title', $data['division']['division_title']);
 
+    $data['main_vote_mp'] = $main_vote_mp;
```
Comment (dracos) [inline www/docs/divisions/division.php]: Not to do with this PR, but if we reversed these if statements and exited if there was no `$vote` or no `$data['division'` we'd have a lot less indenting.

File: www/includes/easyparliament/templates/html/mp/_division_description.php [hunk 2]
```diff
@@ -1,8 +1,6 @@
 <li id="<?= $division['division_id'] ?>" class="<?= $division['strong'] || $show_all ? 'policy-vote--major' : 'policy-vote--minor' ?>">
     <span class="policy-vote__date">On <?= strftime('%e %b %Y', strtotime($division['date'])) ?>:</span>
     <span class="policy-vote__text"><?= $full_name ?><?= $division['text'] ?></span>
-    <?php if ( $division['url'] ) { ?>
-        <a class="vote-description__source" href="<?= $division['url'] ?>">Show full debate</a>
-    <?php } ?>
+    <a class="vote-description__source" href="/divisions/division.php?vote=<?= $division['division_id'] ?>&p=<?= $person_id ?>">Show vote</a>
```
Comment (dracos) [inline www/includes/easyparliament/templates/html/mp/_division_description.php]: Can we have slightly nicer URLs than these? `/divisions/ID` if the ID is unique, I guess?

File: www/includes/easyparliament/templates/html/divisions/vote.php [hunk 3]
```diff
@@ -79,16 +108,22 @@
...
              <ul class="debate-speech__meta debate-speech__links">
+             <?php if ( $division['debate_url'] ) { ?>
                   <li class="link-to-speech">
                       <a class="link debate-speech__meta__link" href="<?= $division['debate_url'] ?>">Show full debate</a>
                   </li>
-            </ul>
             <?php } ?>
+            <?php if ($member && isset($mp_vote)) { ?>
+                <li>
+                    <a class="internal-link debate-speech__meta__link" href="<?= $member->url() ?>/votes"><?= $member->full_name() ?>'s full voting record</a>
```
Comment (dracos) [inline www/includes/easyparliament/templates/html/divisions/vote.php]: Curly apostrophe :)

File: www/includes/easyparliament/templates/html/mp/_division_description.php (hunk 2 above)
Comment (struan) [inline www/includes/easyparliament/templates/html/mp/_division_description.php]: done.

File: www/includes/easyparliament/templates/html/divisions/vote.php (hunk 3 above)
Comment (struan) [inline www/includes/easyparliament/templates/html/divisions/vote.php]: oops, yes.

**Comment by struan** [comment]
This is also ready for a second look.

**Pull request merged**